### PR TITLE
The return of standard cyborgs (sort of)

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Reverts https://github.com/tgstation/tgstation/pull/53276, **but makes standard cyborgs admin-only**. The standard model has been updated to fit the current cyborg mechanics.

Changes the default state of the configuration option for disabling the secborg module to uncommented, meaning that **under the default configuration settings, the security model cannot be selected.** This is to reflect the fact that our actual servers have had secborgs disabled for years and that this is the expected state for this configuration option. **If the maintainers or the headmins desire it, I can atomize this into its own PR.**

The pickaxes of standard cyborgs are now a subtype (with a special description) that has TRAIT_NODROP like the drills of mining cyborgs do.

## Why It's Good For The Game

Admins might want to spawn in standard cyborgs in for TC trades, sentimental reasons, jokes, history lessons, etc., like they currently do sometimes with secborgs.

Private server owners who don't know how to code or how to update the standard model to the current cyborg mechanics might wish to readd standard cyborgs to their server(s).

Standard cyborgs do not require much effort to maintain, as their only unique modules are their epinephrine injectors (which are subtypes of the ones used by mediborgs) and their pickaxes (which I've updated in this PR, and which can be replaced by the mining cyborg's drill in the future if necessary).

Standard cyborgs are a piece of our server's history, and I'd rather not see them be swept under the rug.

## Changelog
:cl:
add: Standard cyborgs have been readded to the game, but are now admin-only.
tweak: The default config settings now disable secborgs from being chosen, to better reflect the config settings that our servers have been operating under for years.
tweak: The standard cyborg's pickaxe now has its own description (and its own name) and TRAIT_NODROP.
/:cl:

### CODE REVIEW

File: code/modules/mining/equipment/mining_tools.dm [hunk 1]
```diff
@@ -63,6 +63,16 @@
...
+/obj/item/pickaxe/cyborg
+	name = "cyborg pickaxe"
+	desc = "An integrated pickaxe."
+
+/obj/item/pickaxe/cyborg/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, CYBORG_ITEM_TRAIT)
+	if(CONFIG_GET(flag/disable_standardborg)) //rip
+		desc = "I used to rule the world..."
+
```
Comment (Fikou) [inline code/modules/mining/equipment/mining_tools.dm:75]: none of this needs to exist, there is nothing different about this pickaxe at all

**Comment by ATH1909** [comment]
Some screenshots from my testing of this PR:
With the config option commented out:
standardborgpick
standardselect
With the config option uncommented:
standardborgpick2
standardselect2
standardupgrade

**Comment by Fikou** [comment]
the changelog is broken

File: code/modules/mining/equipment/mining_tools.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/mining/equipment/mining_tools.dm:75]: why does the mining cyborg's drill get its own subtype, then?

also, I think my joke is funny.

**Comment by ATH1909** [comment]
> the changelog is broken

How do I fix it?

**Comment by Fikou** [comment]
it needs a changelog emote on both sides stinky

File: code/modules/mining/equipment/mining_tools.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/mining/equipment/mining_tools.dm:75]: you should feel free to remove it too

**Comment by ATH1909** [comment]
> it needs a changelog emote on both sides stinky

thanks

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 2]
```diff
@@ -513,6 +513,29 @@
...
+		/obj/item/weldingtool/largetank/cyborg,
+		/obj/item/wrench/cyborg,
+		/obj/item/crowbar/cyborg,
+		/obj/item/stack/sheet/metal,
+		/obj/item/stack/rods/cyborg,
+		/obj/item/stack/tile/plasteel,
+		/obj/item/extinguisher,
+		/obj/item/pickaxe/cyborg,
+		/obj/item/t_scanner/adv_mining_scanner,
+		/obj/item/restraints/handcuffs/cable/zipties,
```
Comment (Fikou) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:532]: WOOOOOOOOOOOO!! YEAAAAAAH BABY THATS WHAT IVE BEEN WAITING FOR THATS WHAT ITS ALL ABOUT WOOOOOOOOO

**Comment by ATH1909** [comment]
If necessary to keep this PR from being closed, I could remove the config parts of it and just make standard cyborgs and secborgs admin-only spawns.

**Comment by tralezab** [comment]
>making it a config option
someone has NOT learned from secborgs. close this pr jesus

**Comment by Qustinnus** [comment]
go away ath

**Comment by Qustinnus** [comment]
also code bloat for no reason other than to move control to the headmins when we have clearly shown we don't want this dumb borg.

**Comment by TemporalOroboros** [comment]
@tralezab I would actually assume that someone has learned perfectly from secborgs... Just not the lesson you wanted them to learn.

**Comment by Jared-Fogle** [comment]
> Private server owners who don't know how to code or how to update the standard model to the current cyborg mechanics might wish to readd standard cyborgs to their server(s).

it's not our problem that the people running fulp apparently are incapable of clicking the revert button

**Comment by Jared-Fogle** [comment]
> And finally, if secborgs deserve to remain in the game as a config option, then by god, standard borgs should deserve that as well. They should be preserved as a reminder of our server's past.

they don't, the pr that removed them entirely (and was correct to do so) caused a big shit storm

**Comment by ATH1909** [comment]
> > Private server owners who don't know how to code or how to update the standard model to the current cyborg mechanics might wish to readd standard cyborgs to their server(s).
> 
> it's not our problem that the people running fulp apparently are incapable of clicking the revert button

Fulp still has standard cyborgs, due to how far behind they are.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I don't see any benefit of this config or not, this is code bloat. We don't want it and downstreams should just code it themselves.

Interested to see if any maintainers disagree. If so, I definitely want this *out* of config, we should be in control of important design decisions wherever possible.

**Comment by Jared-Fogle** [comment]
> Fulp still has standard cyborgs, due to how far behind they are.

So then who is the "downstreams can't code" comment for? We don't design for downstreams, our code is just rather good for modding.

**Comment by Jared-Fogle** [comment]
image

**Comment by ATH1909** [comment]
I've removed the config option part of this PR, per mothblocks's request.

**Comment by Qustinnus** [comment]
image

**Comment by 81Denton** [comment]
> I don't see any benefit of this config or not, this is code bloat. We don't want it and downstreams should just code it themselves.
>
> Interested to see if any maintainers disagree. If so, I definitely want this _out_ of config, we should be in control of important design decisions wherever possible.

I agree, this is code bloat and we don't need it.

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=rz6jEv4IzSs

**Comment by JohnFulpWillard** [comment]
> > Private server owners who don't know how to code or how to update the standard model to the current cyborg mechanics might wish to readd standard cyborgs to their server(s).
> 
> it's not our problem that the people running fulp apparently are incapable of clicking the revert button

We can, we just don’t want it, either

**Pull request closed without merging**